enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+The community put a lot of energy into introducing the [mechanism](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/2392-cloud-controller-manager) to allow cloud-providers and community members
+to build, develop, test and release provider-specific components independently from the Kubernetes core.
+
+Speaking about the Openshift integrations with cloud providers - for the moment, a lot of things are tended to be encoded
+in [the Openshift codebase](https://docs.providers.openshift.org/cloud-controller-manager/#integrating-with-other-openshift-components)
+within API definitions, operators logic, and installer program code.
+This fact creates quite  a lot of obstacles for RH partners and community members in their attempts to add new cloud providers to Openshift,
+as well as making RH engineering involvement quite necessary.
+
+Lately, there have been several initiatives around making Openshift more composable and flexible,
+for example [Capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md) and
+[Platform Operators](https://github.com/openshift/enhancements/blob/master/enhancements/olm/platform-operators.md) are
+the significant steps towards this direction. However, despite having these powerful instruments, there is still necessary
+to land a code into the Openshift codebase for technical enablement of a new cloud provider, which might be hard or
+nearly impossible for external contributors.
+
+This EP describes adding a new special platform type, referred here as "External".
+
+Imagine some "MomAndPop Cloud" have created an infrastructure platform that resembles AWS but has its own API that is different than AWS.
+They would like to give their users the best OpenShift experience possible, but integrating their code into a Red Hat release is not possible for them.
+Using the "External" platform, [capabilities](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md),
+and [platform operators](https://github.com/openshift/enhancements/blob/master/enhancements/olm/platform-operators.md), they can still deliver this functionality by creating their own Cloud Controller Managers,
+CSI drivers, network topology operators, Machine API controllers, and OpenShift configurations.
+This allows "MomAndPop Cloud" to provide the best OpenShift experience while also developing their components
+without the necessity of tying to Red Hat's internal processes or keeping a fork of the significant part of the Openshift code base.
+
+The new "External" platform type would serve as a signal when OpenShift is running on an infrastructure platform
+that will receive supplemental components from the infrastructure provider.
+
+"External" is different from our current platform types in that it will signal that the infrastructure
+is specifically not "None" or any of the known providers (e.g., AWS, GCP, etc.).
+
+## Motivation
```

Comment:
In establishing the motivation, it may help to emphasize that we're designing for a world where a large number of regional cloud providers will take advantage of this new ability.

Reply:
I tried to reword one of paragraphs within this section, hope it helped